numpy/ma/core.py

```diff
@@ -711,14 +716,32 @@ def getdata(a, subok=True):
     array([[1, 2],
            [3, 4]])
 
+    When `dtype` is specified, this is used to process ambiguous objects:
+
+    >>> ma.getdata((1, 2, 3))
+    array([1, 2, 3])
+    >>> ma.getdata((1, 2, 3), dtype=object)
+    (1, 2, 3)
+    >>> ma.getdata((1, 2, 3), dtype=[('a', int), ('b', int), ('c', int)])
+    array((1, 2, 3), 
+      dtype=[('a', '<i4'), ('b', '<i4'), ('c', '<i4')])
     """
+    # first deal with masked arrays
     try:
         data = a._data
     except AttributeError:
-        data = np.array(a, copy=False, subok=subok)
```

Comment:
Is there a particular reason not to keep the logic here? That avoids having to add the `else` clause.